### PR TITLE
tmz 3.1.0 feature battleship

### PR DESCRIPTION
**Pull request by t-m-z**

### CODE REVIEW

File: lib/tools/crypto_and_encodings/battleship/logic/battleship_data.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
+part of 'package:gc_wizard/tools/crypto_and_encodings/battleship/logic/battleship.dart';
+
+const BATTLESHIP_ERROR_INVALID_PAIR = 'battleship_error_invalid_pair';
+const BATTLESHIP_EMPTY_LINE = '                                                           ';
+final Map<String, List<String>> BATTLESHIP_ALPHABET = {
```
Comment (Mike-3) [inline lib/tools/crypto_and_encodings/battleship/logic/battleship_data.dart:5]: Change to const

**Comment by Mike-3** [comment]
image

Exception

**Review by Mike-3 — CHANGES_REQUESTED** [review]

File: lib/tools/crypto_and_encodings/battleship/logic/battleship_data.dart (hunk 1 above)
Comment (t-m-z) [reply lib/tools/crypto_and_encodings/battleship/logic/battleship_data.dart]: done

Comment (t-m-z) [reply lib/tools/crypto_and_encodings/battleship/logic/battleship_data.dart]: exception caught, tests will follow

**Comment by Mike-3** [comment]
grafik

Out of Memory Exception -> Limitation

**Comment by t-m-z** [comment]
valid point. I am still not experienced enough in testing and I still believe the users are good

**Comment by Mike-3** [comment]
Mayby the user, but not the tester ;-)
Don't forget to change to default decrypt.

**Pull request merged**